diff -uNdr boost_1_59_0/libs/exception/test/cloning_test.cpp vx7/workspace/VSBSim32/usr/src/BOOST/boost_1_59_0/libs/exception/test/cloning_test.cpp
--- boost_1_59_0/libs/exception/test/cloning_test.cpp	2014-08-29 17:51:11.000000000 -0700
+++ boost_1_59_0/libs/exception/test/cloning_test.cpp	2016-03-11 21:56:23.072553959 -0800
@@ -77,6 +77,9 @@
         {
         return wh_;
         }
+#if  defined(__VXWORKS__) && (__GNUC__ == 4 && __GNUC_MINOR__ < 9)     
+    ~derives_std_boost_exception() throw() {};
+#endif
     };
 
 struct
diff -uNdr boost_1_59_0/libs/exception/test/diagnostic_information_test.cpp vx7/workspace/VSBSim32/usr/src/BOOST/boost_1_59_0/libs/exception/test/diagnostic_information_test.cpp
--- boost_1_59_0/libs/exception/test/diagnostic_information_test.cpp	2014-08-29 17:51:11.000000000 -0700
+++ boost_1_59_0/libs/exception/test/diagnostic_information_test.cpp	2016-03-11 21:50:15.443456682 -0800
@@ -32,6 +32,9 @@
         {
         return "error1";
         }
+#if  defined(__VXWORKS__) && (__GNUC__ == 4 && __GNUC_MINOR__ < 9)      
+    ~error1() throw() {};
+#endif    
     };
 
 struct
@@ -61,6 +64,9 @@
         {
         return diagnostic_information_what(*this);
         }
+#if  defined(__VXWORKS__) && (__GNUC__ == 4 && __GNUC_MINOR__ < 9)      
+    ~error4() throw() {};
+#endif    
     };
 
 void
diff -uNdr boost_1_59_0/libs/exception/test/errinfos_test.cpp vx7/workspace/VSBSim32/usr/src/BOOST/boost_1_59_0/libs/exception/test/errinfos_test.cpp
--- boost_1_59_0/libs/exception/test/errinfos_test.cpp	2014-08-29 17:51:11.000000000 -0700
+++ boost_1_59_0/libs/exception/test/errinfos_test.cpp	2016-03-11 21:50:18.932343722 -0800
@@ -22,6 +22,9 @@
     virtual boost::exception,
     virtual std::exception
     {
+#if  defined(__VXWORKS__) && (__GNUC__ == 4 && __GNUC_MINOR__ < 9)      
+    ~test_exception() throw() {};
+#endif    
     };
 
 int
diff -uNdr boost_1_59_0/libs/exception/test/Jamfile.v2 vx7/workspace/VSBSim32/usr/src/BOOST/boost_1_59_0/libs/exception/test/Jamfile.v2
--- boost_1_59_0/libs/exception/test/Jamfile.v2	2014-08-29 17:51:11.000000000 -0700
+++ boost_1_59_0/libs/exception/test/Jamfile.v2	2016-03-09 15:33:57.461439653 -0800
@@ -38,7 +38,8 @@
 run diagnostic_information_test.cpp ;
 run refcount_ptr_test.cpp ;
 run current_exception_cast_test.cpp ;
-run no_exceptions_test.cpp : : : <exception-handling>off ;
+run no_exceptions_test.cpp : : : <exception-handling>off 
+								<target-os>vxworks:<build>no ; # vx requires complete library rebuild to turn off exceptions 
 run errinfos_test.cpp ;
 run exception_ptr_test.cpp/<define>BOOST_ENABLE_NON_INTRUSIVE_EXCEPTION_PTR ../../thread/src/tss_null.cpp /boost/exception /boost//thread : : : <threading>multi : non_intrusive_exception_ptr_test ;
 run exception_ptr_test.cpp ../../thread/src/tss_null.cpp /boost//thread : : : <threading>multi ;
diff -uNdr boost_1_59_0/libs/exception/test/N3757_test.cpp vx7/workspace/VSBSim32/usr/src/BOOST/boost_1_59_0/libs/exception/test/N3757_test.cpp
--- boost_1_59_0/libs/exception/test/N3757_test.cpp	2014-08-29 17:51:11.000000000 -0700
+++ vx7/workspace/VSBSim32/usr/src/BOOST/boost_1_59_0/libs/exception/test/N3757_test.cpp	2016-03-11 21:50:21.838249638 -0800
@@ -27,7 +27,11 @@
     return s << "my_error_code(" << x.ec << ')';
 }
 
-struct my_exception: virtual std::exception, virtual boost::exception { };
+struct my_exception: virtual std::exception, virtual boost::exception {
+#if  defined(__VXWORKS__) && (__GNUC__ == 4 && __GNUC_MINOR__ < 9)      
+    ~my_exception() throw() {};
+#endif    
+};
 
 int
 main()
diff -uNdr boost_1_59_0/libs/exception/test/throw_exception_test.cpp vx7/workspace/VSBSim32/usr/src/BOOST/boost_1_59_0/libs/exception/test/throw_exception_test.cpp
--- boost_1_59_0/libs/exception/test/throw_exception_test.cpp	2014-08-29 17:51:11.000000000 -0700
+++ boost_1_59_0/libs/exception/test/throw_exception_test.cpp	2016-03-11 21:50:24.417166140 -0800
@@ -22,6 +22,9 @@
     std::exception,
     boost::exception
     {
+#if  defined(__VXWORKS__) && (__GNUC__ == 4 && __GNUC_MINOR__ < 9)      
+    ~exception2() throw() {};
+#endif
     };
 
 void
