diff -uNdr boost_1_59_0.orig/boost/random/mersenne_twister.hpp boost_1_59_0/boost/random/mersenne_twister.hpp
--- boost_1_59_0.orig/boost/random/mersenne_twister.hpp	2015-02-27 09:44:45.000000000 -0800
+++ boost_1_59_0/boost/random/mersenne_twister.hpp	2017-02-25 16:40:22.476275925 -0800
@@ -149,7 +149,7 @@
         for (i = 1; i < n; i++) {
             // See Knuth "The Art of Computer Programming"
             // Vol. 2, 3rd ed., page 106
-            x[i] = (f * (x[i-1] ^ (x[i-1] >> (w-2))) + i) & mask;
+            x[i] = (f * (x[i-1] ^ (x[i-1] >> (static_cast<UIntType>(w)-2))) + static_cast<UIntType>(i)) & mask;
         }
 
         normalize_state();
